pkg/tools/socket.go

```diff
@@ -0,0 +1,135 @@
+package tools
+
+import (
+	"context"
+	"github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc"
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"net"
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	opentracingEnv     = "OPEN_TRACING"
+	opentracingDefault = true
+	insecureEnv        = "INSECURE"
+	insecureDefault    = true
```

Comment:
This is a fine default for the moment... but we will eventually want to default to secure.

Reply:
Sure, I will change values by default during rebase of my Security PR